keps/draft-0003-kep-cli.md

```diff
@@ -0,0 +1,133 @@
+      * [Summary](#summary)
+      * [Motivation](#motivation)
+         * [Goals](#goals)
+         * [Non-Goals](#non-goals)
+      * [Proposal](#proposal)
+         * [User Stories](#user-stories)
+            * [Installing a framework](#installing-a-framework)
+            * [Uninstalling a framework](#uninstalling-a-framework)
+            * [Listing running frameworks](#listing-running-frameworks)
+            * [Getting the status of a running framework](#getting-the-status-of-a-running-framework)
+            * [Start specific plans](#start-specific-plans)
+            * [Get the history of planexecutions](#get-the-history-of-planexecutions)
+            * [Executing into a particular framework](#executing-into-a-particular-framework)
+            * [Run framework specific commands](#run-framework-specific-commands)
+            * [Shell into containers of a framework](#shell-into-containers-of-a-framework)
+            * [Read all combined logs of a framework](#read-all-combined-logs-of-a-framework)
+         * [Implementation Details/Notes/Constraints](#implementation-detailsnotesconstraints)
+         * [Risks and Mitigations](#risks-and-mitigations)
+      * [Graduation Criteria](#graduation-criteria)
+      * [Implementation History](#implementation-history)
+
+## Summary
+
+This KEP aims to improve the end user experience via a CLI that makes the life of an operator easier. Right now,
+interacting with KUDO consists of a chain of cumbersome steps that require specific business logic. By providing one
+global entry point for the user (e.g. to install or interact with frameworks, get the status of current plans etc.) we
+can abstract complexity from the user and drive adaptation.
+
+## Motivation
+
+Currently, installing your `Frameworks` requires not just the proper configuration of your YAML files but also multiple 
+interactions with `kubectl`. Having CLI tooling to simplify the path for cluster operators to deployment and maintenance 
+ of installed `Frameworks` adds not just value across the community but also helps implementing best practices when using KUDO.
+
+### Goals
+
+The goal of this KEP is to have a CLI binary that can be used under the Kubectl Plugin System to work with KUDO.
+
+- We should drive making the adaption to KUDO easier
+- Not to confuse flags
+- Providing metrics for usage and error reporting
+
+### Non-Goals
+
+Non-Goals are:
+
+- Just building another wrapper around `kubectl`
+- Adding complexity
+
+## Proposal
+
+This is where we get down to the nitty gritty of what the proposal actually is.
+
+### User Stories
+
+#### Installing a framework
+
+As a user I want to be able to install a framework to my cluster.
+
+#### Uninstalling a framework
+
+As a user I want to be able to uninstall a framework from my cluster.
+
+#### Listing running frameworks
+
+As a user I want to be able to list current installed frameworks in my namespace.
+
+#### Getting the status of a running framework
```

Comment:
You mean Instance?

Reply:
+1

Reply:
Yes, and very good find! I see this important with `kep-2` coming along to not confuse `Framework` with framework or `Component`. I had the fear that once we have dynamic CRDs the definition of Framework wouldn't apply here anymore and we would have to rewrite. Originally it was intended as _running framework_ aka that would be an Instance. 

I also think the user story should be getting the status of an Instance, leaving the `running` entirely out.

Reply:
The `Components` in that branch are just a playground, those won't be merged in with those names.